app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(
+            self,
+            name: str,
+            age: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_instances = {}
+    for person_info in people:
+        person_name = person_info["name"]
+        person = Person(person_name, person_info["age"])
+        person_instances[person_name] = person
+
+    for person_info in people:
+        person_name = person_info["name"]
```

Comment:
Retrieve not a person name, but a person instance here
```suggestion
        person_instance = Person.people[person_info["name"]]
```